haystack/dataclasses/streaming_chunk.py

```diff
@@ -102,6 +102,71 @@ def __post_init__(self):
         if (self.tool_calls or self.tool_call_result) and self.index is None:
             raise ValueError("If `tool_call`, or `tool_call_result` is set, `index` must also be set.")
 
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Returns a dictionary representation of the StreamingChunk.
+
+        :returns: Serialized dictionary representation of the calling object.
+        """
+        return {
+            "content": self.content,
+            "meta": self.meta,
+            "component_info": asdict(self.component_info) if self.component_info else None,
```

Comment:
Instead of this could we add `to_dict` and `from_dict` methods to `ComponentInfo` and then call them here. This way we also can serialize/deserialize each of our other dataclasses as well independently. 

So this line would then become
```suggestion
            "component_info": self.component_info.to_dict() if self.component_info else None,
```

Reply:
Done, thank you!